app/models/nil_site.rb

```diff
@@ -0,0 +1,59 @@
+# NilSite is used to represent the Site in the global tenant in a multitenant environment
+# (i.e. Sites only exist on individual tenants and never globally)
+class NilSite
+  class << self
+    # NilSite must be a singleton like Site
+    attr_writer :instance
+    def instance
+      @instance ||= NilSite.new
+    end
+  end
+
+  def id
```

Comment:
You can replace all this `nil`-returning methods with a single `attr_reader` declaration.